File: Easy/2965. Find Missing and Repeated Values/Go/main.go
func findMissingAndRepeatedValues(grid [][]int) []int {
  n := len(grid)
  count := make(map[int]int)
  for i := 0; i < n; i++ {
      for j := 0; j < n; j++ {
          count[grid[i][j]]++
      }
  }
  repeated, missing := 0, 0
  for i := 1; i <= n*n; i++ {
      if count[i] == 2 {
          repeated = i
      } else if count[i] == 0 {
          missing = i
      }
  }
  return []int{repeated, missing}
}

File: Easy/2965. Find Missing and Repeated Values/Rust/main.rs
impl Solution {
    pub fn find_missing_and_repeated_values(grid: Vec<Vec<i32>>) -> Vec<i32> {
        let n = grid.len();
        let mut counts = vec![0; n * n + 1];
        let mut repeated = 0;
        let mut missing = 0;

        for row in grid.iter() {
            for &val in row.iter() {
                counts[val as usize] += 1;
            }
        }

        for i in 1..=n * n {
            if counts[i] == 0 {
                missing = i as i32;
            } else if counts[i] > 1 {
                repeated = i as i32;
            }
        }

        vec![repeated, missing]

    }
}
